test/gpu/native/studies/sort/radixSortGpu.chpl

```diff
@@ -0,0 +1,279 @@
+module RadixSort {
+  import BitOps;
+  use GPU;
+
+  config const noisy = false;
+  config const scanTest = false;
+  config const scanType = "blelloch"; // "serial", "hillisSteele", "blelloch"
+  config const bitsAtATime = 4; // 1-8 are all reasonable values
+                                // but for some reason 16 performs best.
+                                // My guess is because we're dealing with
+                                // uint(64) it makes sense for 16 to do well
+                                // since its a nice even factor and causes
+                                // only 4 iterations of a single radix sort step
+  // How many bits we sort at once based on the bitsAtATime
+  const buckets = 1 << bitsAtATime; // 2^bitsAtATime, ex: 2^4 = 16 = 0b10000
+  const bitMask = buckets - 1; // 2^bitsAtATime - 1, ex: 2^4 - 1 = 15 = 0b1111
+
+  proc serialScan(ref prefixSum: [] uint, buckets: int){
+    // Convert this count array into a prefix sum
+    // This is the same as the count array, but each element is the sum of all previous elements
+    // This is an exclusive scan
+    // Serial implementation
+
+    // Calculate the prefix sum
+    var sum : uint = 0;
+    for i in 0..<buckets {
+      var t : uint = prefixSum[i];
+      prefixSum[i] = sum;
+      sum += t;
+    }
+  }
+
+  proc hillisSteeleScan(ref hostPrefixSum: [] uint, hostBuckets: int){
+    // Hillis Steele Scan
+    // This is the same as the count array, but each element is the sum of all previous elements
+    // Uses a naive algorithm that does O(nlogn) work
+    // Hillis and Steele (1986)
+    on here.gpus[1]{
+      var buckets = hostBuckets;
+      var prefixSum = hostPrefixSum;
+      var offset = 1;
+      while offset <= buckets {
+          var prefixSumBuffer = prefixSum;
+          forall i in offset..<buckets { // foreach causes error? Investigate later
```

Comment:
If I use `foreach` here's what I get:
```chpl
internal error: gpu-nvidia.c:240: Error calling CUDA function: invalid argument (Code: 1)
```

Reply:
If anything, your response makes me more worried. Does it have something to do with `..<` which may not be something we test with.

Reply:
Just checked by replacing the bounds with `offset..(buckets-1)` and used a `foreach` loop and still got the same error. Which is not surprising because I use the `..<` pattern in a `foreach` loop in the `blellochScan` function and that works fine, so something else must be causing the error.

Reply:
OK, thanks for checking. It'd be great to put some time into this to understand the issue. Can I ask you to create a private issue to remind us of this? This is a very innocent-looking loop and discrepancy between foreach/forall is worrisome.

Reply:
So it turns out that error was a bug in my code.
The conditional for the `while` loop on line 42 should be `offset<buckets`.
But it's still weird that it only fails when I use a `foreach` loop, so I created an issue: https://github.com/chapel-lang/chapel/issues/23095